cocos/2d/CCNode.cpp

```diff
@@ -1761,6 +1761,25 @@ AffineTransform Node::getNodeToParentAffineTransform() const
     return ret;
 }
 
+
+Mat4 Node::getNodeToParentTransform(Node* ancestor) const
+{
+    Mat4 t(this->getNodeToParentTransform());
+
+    for (Node *p = _parent;  p != nullptr && p != ancestor ; p = p->getParent())
+    {
+        t = p->getNodeToParentTransform() * t;
+    }
+
+    return t;
+}
+
+AffineTransform Node::getNodeToParentAffineTransform(Node* ancestor) const
+{
+    AffineTransform ret;
+    GLToCGAffine(getNodeToParentTransform(ancestor).m, &ret);
```

Comment:
I think this implementation is not alright, please refer to `getNodeToWorldAffineTransform` you need to concat affine transform level by level